core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,178 @@
+)
+
+// Log Event Trigger Capability Request Config Details
+type RequestConfig struct {
+	ContractName         string                     `json:"contractName"`
+	ContractAddress      common.Address             `json:"contractAddress"`
+	ContractEventName    string                     `json:"contractEventName"`
+	ContractReaderConfig evmtypes.ChainReaderConfig `json:"contractReaderConfig"`
+}
+
+// LogEventTrigger struct to listen for Contract events using ContractReader gRPC client
+// in a loop with a periodic delay of pollPeriod milliseconds, which is specified in
+// the job spec
+type logEventTrigger struct {
+	ch   chan<- capabilities.TriggerResponse
+	lggr logger.Logger
+	ctx  context.Context
+
+	// Contract address and Event Signature to monitor for
+	reqConfig      *RequestConfig
+	contractReader types.ContractReader
+	relayer        core.Relayer
+	startBlockNum  uint64
+
+	// Log Event Trigger config with pollPeriod and lookbackBlocks
+	logEventConfig LogEventConfig
+	ticker         *time.Ticker
+	done           chan bool
+}
+
+// Construct for logEventTrigger struct
+func newLogEventTrigger(ctx context.Context,
+	reqConfig *RequestConfig,
+	logEventConfig LogEventConfig,
+	relayer core.Relayer) (*logEventTrigger, chan capabilities.TriggerResponse, error) {
+	jsonBytes, err := json.Marshal(reqConfig.ContractReaderConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create a New Contract Reader client, which brings a corresponding ContractReader gRPC service
+	// in Chainlink Core service
+	contractReader, err := relayer.NewContractReader(ctx, jsonBytes)
+	if err != nil {
+		return nil, nil,
+			fmt.Errorf("error fetching contractReader for chainID %d from relayerSet: %v", logEventConfig.ChainId, err)
+	}
+
+	// Bind Contract in ContractReader
+	boundContracts := []types.BoundContract{{Name: reqConfig.ContractName, Address: reqConfig.ContractAddress.Hex()}}
+	err = contractReader.Bind(ctx, boundContracts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Get current block HEAD/tip of the blockchain to start polling from
+	latestHead, err := relayer.LatestHead(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting latestHead from relayer client: %v", err)
+	}
+	height, err := strconv.ParseUint(latestHead.Height, 10, 64)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid height in latestHead from relayer client: %v", err)
+	}
+
+	// Setup callback channel, logger and ticker to poll ContractReader
+	callbackCh := make(chan capabilities.TriggerResponse, defaultSendChannelBufferSize)
+	ticker := time.NewTicker(time.Duration(logEventConfig.PollPeriod) * time.Millisecond)
+	done := make(chan bool)
+	lggr, err := logger.New()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not initialise logger for LogEventTrigger")
+	}
+
+	// Initialise a Log Event Trigger
+	l := &logEventTrigger{
+		ch:   callbackCh,
+		lggr: logger.Named(lggr, "LogEventTrigger: "),
```

Comment:
Why the `: `?  All the other logger.Named I see have no `: ` suffix.  

Also, is there any desire to include part of the config in the name?  Elsewhere I see `	log := d.logger.Named("StandardCapabilities").Named(spec.StandardCapabilitiesSpec.GetID())
`

Reply:
Sure, I'll remove this